client/core/trade.go

```diff
@@ -1064,7 +1064,7 @@ func (t *trackedTrade) shouldBeginFindRedemption(ctx context.Context, match *mat
 		return false
 	}
 
-	confs, spent, err := t.wallets.fromWallet.SwapConfirmations(ctx, swapCoinID, proof.Script, match.MetaData.Stamp)
+	confs, spent, err := t.wallets.fromWallet.SwapConfirmations(ctx, swapCoinID, proof.Script, match.MetaData.Stamp, t.metaData.FromVersion)
```

Comment:
Presently eth's `(*swapReceipt).Contract() dex.Bytes` returns a slice that contains the secret hash.

What if it returned a slice that was a concatenation of contract version and secret hash, so that `client.asset.Receipt.Contract()`, which is assigned to `proof.Script`, has this info.  It's already not a script at all as implied by the `MatchProof.Script` field name, and it's intended to be an opaque blob for multi-asset support, so why not have it encode the version too?

That is, we're already freely reinterpreting the `Receipt.Contract` method, so I'd just assume we use it to convey all the info needed about this contract/swap rather than add an arg to `SwapConfirmations`, which without close examination is kind of a head-scratcher.